presto-base-jdbc/src/test/java/com/facebook/presto/plugin/jdbc/optimization/TestJdbcJoinPushdown.java

```diff
@@ -0,0 +1,280 @@
+        TableHandle tableHandle = new TableHandle(new ConnectorId(CATALOG_NAME), connectorTableHandle, new ConnectorTransactionHandle() {}, Optional.of(jdbcTableLayoutHandle));
+
+        return PLAN_BUILDER.tableScan(
+                tableHandle,
+                Arrays.stream(columnNames).map(column -> newVariable(column, type)).collect(toImmutableList()),
+                Arrays.stream(columnNames)
+                        .map(column -> newVariable(column, type))
+                        .collect(toMap(identity(), entry -> getColumnHandleForVariable(entry.getName(), type))));
+    }
+
+    private static void assertPlanMatch(PlanNode actual, PlanMatchPattern expected)
+    {
+        assertPlanMatch(actual, expected, TypeProvider.empty());
+    }
+
+    private static void assertPlanMatch(PlanNode actual, PlanMatchPattern expected, TypeProvider typeProvider)
+    {
+        PlanAssert.assertPlan(
+                TEST_SESSION,
+                METADATA,
+                (node, sourceStats, lookup, session, types) -> PlanNodeStatsEstimate.unknown(),
+                new Plan(actual, typeProvider, StatsAndCosts.empty()),
+                expected);
+    }
+
+    private static final class JdbcTableScanMatcher
+            implements Matcher
+    {
+        private final JdbcTableLayoutHandle jdbcTableLayoutHandle;
+        private final Set<ColumnHandle> columns;
+
+        static PlanMatchPattern jdbcTableScanPattern(JdbcTableLayoutHandle jdbcTableLayoutHandle, Set<ColumnHandle> columns)
+        {
+            return node(TableScanNode.class).with(new TestJdbcJoinPushdown.JdbcTableScanMatcher(jdbcTableLayoutHandle, columns));
+        }
+
+        private JdbcTableScanMatcher(JdbcTableLayoutHandle jdbcTableLayoutHandle, Set<ColumnHandle> columns)
+        {
+            this.jdbcTableLayoutHandle = jdbcTableLayoutHandle;
+            this.columns = columns;
+        }
+
+        @Override
+        public boolean shapeMatches(PlanNode node)
+        {
+            return node instanceof TableScanNode;
+        }
+
+        @Override
+        public MatchResult detailMatches(PlanNode node, StatsProvider stats, Session session, Metadata metadata, SymbolAliases symbolAliases)
+        {
+            checkState(shapeMatches(node), "Plan testing framework error: shapeMatches returned false in detailMatches in %s", this.getClass().getName());
+
+            TableScanNode tableScanNode = (TableScanNode) node;
+            JdbcTableLayoutHandle layoutHandle = (JdbcTableLayoutHandle) tableScanNode.getTable().getLayout().get();
+
+            if (jdbcTableLayoutHandle.getTable().equals(layoutHandle.getTable())
+                    && jdbcTableLayoutHandle.getTupleDomain().equals(layoutHandle.getTupleDomain())
+                    && ((!jdbcTableLayoutHandle.getAdditionalPredicate().isPresent() && !layoutHandle.getAdditionalPredicate().isPresent())
```

Comment:
[additionalPredicate](https://github.com/prestodb/presto/blob/70f98518ac8080a3406b461d5ac534f19738ddc5/presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcTableLayoutHandle.java#L102) direct comparison should work fine ?